monitor.h

```diff
@@ -1,3 +1,66 @@
-#pragma once
+#ifndef MONITOR_H_
+#define MONITOR_H_
 
-int vitalsOk(float temperature, float pulseRate, float spo2);
+#include <string>
+#include <vector>
+#include <tuple>
+#include "./message_handler.h"
+#include "./temperature_converter.h"
+
+// Enum for Vital alert levels
+enum class VitalAlertLevel {
+    OK,
+    WARNING,
+    CRITICAL
+};
+
+class Monitor {
+ private:
+    std::string tempUnit;
+    MessageHandler msgHandler;
+    TemperatureConverter tempConverter;
+
+    // Unified function to check if a value is within limits
+    VitalAlertLevel checkValue(float value, float lowerLimit, float upperLimit);
+
+    // Function to check temperature specifically
+    VitalAlertLevel checkTemperature(float temperature);
+
+    // Function to display the temperature with the appropriate unit
+    std::string formatTemperature(float temperature);
+
+    // Helper function to handle critical alert animation
+    void displayCriticalAlertAnimation();
+
+    // Function to display alert level
+    void displayAlertLevel(const std::string& message, VitalAlertLevel level);
+
+    // Handle individual vital checks and return the appropriate level
+    VitalAlertLevel checkVital(float value, float lowerLimit, float upperLimit,

```

Comment:
Purpose of `checkValue` vs `checkVital` isn't clear, unless we see the code inside them.
Try renaming by their purpose.
Hint: Avoid the use of the word `check`. And try re-organizing the function if it makes it easier to name it.